subsys/bluetooth/controller/ll_sw/ull_llcp_pdu.c

```diff
@@ -173,11 +177,11 @@ void ull_cp_priv_pdu_decode_feature_rsp(struct ull_cp_conn *conn,
 {
 	u64_t featureset;
 
-
 	feature_filter(pdu->llctrl.feature_rsp.features, &featureset);
+	conn->llcp.fex.features_used = LL_FEAT & featureset;
 
-	conn->llcp.fex.features = featureset;
-
+	featureset &= (FEAT_FILT_OCTET0 | conn->llcp.fex.features_used);
```

Comment:
Isn't the octet 0 filtering actually pointless at this point, the RSP PDU FeatureSet should contain FeatureSet_A[1..7] + (FeatureSet_A[0] & FeatureSet_B[0]) where A/B = S/M for master initiated and M/S for slave initiated. Or am i reading this wrong?

Reply:
For a well-behaving peer you are right; I am here allowing for a misbehaving peer, but maybe I am too paranoid?

Reply:
Ah, yes that makes sense. Keep the code.

Reply:
On further consideration, even with a misbehaving peer we'll be doing the *right* thing given the functions we've defined for checking features